src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1593,6 +1593,87 @@ public void MainWindowHandle_GetWithGui_ShouldRefresh_Windows()
             }
         }
 
+        [Fact]
+        [OuterLoop]
+        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Pops UI
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void MainWindowTitle_GetWithGui_ShouldRefresh_Windows()
+        {
+            const string ExePath = "notepad.exe";
+            Assert.True(IsProgramInstalled(ExePath));
+
+            using (Process process = Process.Start(ExePath))
+            {
+                try
+                {
+                    Assert.Equal(string.Empty, process.MainWindowTitle);
+
+                    for (int attempt = 0; attempt < 50; ++attempt)
+                    {
+                        process.Refresh();
+                        if (process.MainWindowTitle != string.Empty)
+                        {
+                            break;
+                        }
+
+                        Thread.Sleep(100);
+                    }
+
+                    Assert.NotEqual(string.Empty, process.MainWindowTitle);
+                }
+                finally
+                {
+                    process.Kill();
+                    Assert.True(process.WaitForExit(WaitInMS));
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Pops UI
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Responding_GetWithGui_ShouldRefresh_Windows()
+        {
+            const string ExePath = @"C:\Program Files\Windows NT\Accessories\wordpad.exe";
+            Assert.True(IsProgramInstalled(ExePath));
+
+            string dummyFilePath = $@"{Path.GetTempPath()}dummy_file";
+
+            var dummyFile = new FileStream(dummyFilePath, System.IO.FileMode.Create);
+            _ = dummyFile.Seek(2048L * 1024 * 1024, SeekOrigin.Begin);
+            dummyFile.WriteByte(0);
+            dummyFile.Close();
```

Comment:
Woh, this is creating a 2GB temporary file?  Even in outerloop that's probably something to be avoided.

Reply:
You're right, the file is quite big.
But I want to make sure that even on the fastest machine, Wordpad goes into a "not responding" state.
If the file is too small, Wordpad will have loaded the file without getting into that state.

So, some kind of optimum file size has to be found.

Reply:
If we really need to test that, I suggest the test itself needs to be the thing that causes the non-responsiveness, e.g. the test would RemoteExecutor.Invoke code that would create a window that's not responsive.  We shouldn't depend on creating a big enough file that the particular implementation in wordpad causes the UI to hang.

Reply:
@stephentoub you mean with something like windows forms? I tried the following
```csharp
static void RunNotRespondingApp()
{
    var form = new System.Windows.Forms.Form();
    form.Show();
    Thread.Sleep(60_000);
}

RemoteExecutor.Invoke(RunNotRespondingApp);
```
but for whatever reason it wasn't enough to simulate not responding state using RemoteExecutor. It does work without issue when I run as a standalone app.